From 01b9c931246e75771ebc499f5659d7dc49aa6233 Mon Sep 17 00:00:00 2001
From: Florin9doi <florin9doi@gmail.com>
Date: Sun, 16 Sep 2018 13:00:44 +0300
Subject: [PATCH 3/4] Merge ignoreCall and endCall buttons

-reuse a single TextView for both buttons: ignoreCall and endCall
-endCall was completely misplaced, this patch resolves the issue
-move the image and text for each button in a new View and move
 onClickListener on the parent. This change allows to accept/ignore
 a phonecall by clicking on the text.

Change-Id: I991a1eba3a3652ad0da933b2ee45d2197ef273fb
---
 res/layout/phone_panel.xml                    | 104 ++++++++----------
 src/org/lineageos/flipflap/CircleView.java    |  25 ++---
 .../lineageos/flipflap/RectangularView.java   |  25 ++---
 3 files changed, 63 insertions(+), 91 deletions(-)

diff --git a/res/layout/phone_panel.xml b/res/layout/phone_panel.xml
index 3caae2f..5648e50 100644
--- a/res/layout/phone_panel.xml
+++ b/res/layout/phone_panel.xml
@@ -33,76 +33,62 @@
         android:layout_below="@id/incoming_call_number"
         android:src="@drawable/ic_ring_icon" />
 
-    <ImageButton
+    <RelativeLayout
         android:id="@+id/answer_button"
-        android:layout_width="30dp"
-        android:layout_height="wrap_content"
-        android:scaleType="fitXY"
-        android:layout_alignParentLeft="true"
-        android:layout_marginLeft="80dip"
-        android:layout_alignBottom="@id/incoming_call_alert_icon"
-        android:src="@drawable/ic_call_answer"
-        android:layout_below="@id/incoming_call_number"
-        android:background="@android:color/transparent" />
-
-    <ImageButton android:id="@+id/ignore_button"
-        android:layout_width="30dp"
-        android:layout_height="wrap_content"
-        android:scaleType="fitXY"
-        android:layout_alignParentRight="true"
-        android:layout_marginRight="80dip"
-        android:layout_alignBottom="@id/incoming_call_alert_icon"
-        android:src="@drawable/ic_call_end"
-        android:layout_below="@id/incoming_call_number"
-        android:background="@android:color/transparent"/>
-
-    <TextView
-        android:id="@+id/answer_text"
-        style="@style/label_bold"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_alignParentLeft="true"
         android:layout_marginLeft="70dip"
-        android:layout_centerVertical="true"
-        android:textColor="@color/clock_white"
-        android:layout_below="@id/incoming_call_alert_icon"
-        android:text="@string/answer_call"
-        android:textAlignment="center" />
+        android:layout_below="@id/incoming_call_number" >
 
-    <TextView
-        android:id="@+id/ignore_text"
-        style="@style/label_bold"
+        <ImageView
+            android:id="@+id/answer_icon"
+            android:layout_width="30dp"
+            android:layout_height="40dp"
+            android:layout_centerHorizontal="true"
+            android:scaleType="fitXY"
+            android:src="@drawable/ic_call_answer"
+            android:background="@android:color/transparent" />
+
+        <TextView
+            android:id="@+id/answer_text"
+            style="@style/label_bold"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_centerVertical="true"
+            android:textColor="@color/clock_white"
+            android:layout_below="@id/answer_icon"
+            android:text="@string/answer_call"
+            android:textAlignment="center" />
+    </RelativeLayout>
+
+    <RelativeLayout
+        android:id="@+id/ignore_button"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_alignParentRight="true"
-        android:layout_centerVertical="true"
         android:layout_marginRight="70dip"
-        android:textColor="@color/clock_white"
-        android:layout_below="@id/incoming_call_alert_icon"
-        android:text="@string/ignore_call"
-        android:textAlignment="center" />
+        android:layout_below="@id/incoming_call_number" >
 
-    <ImageButton android:id="@+id/end_call_button"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_centerVertical="true"
-        android:src="@drawable/ic_call_end"
-        android:layout_below="@id/incoming_call_alert_icon"
-        android:background="@android:color/transparent"
-        android:visibility="invisible" />
+        <ImageView
+            android:id="@+id/ignore_icon"
+            android:layout_width="30dp"
+            android:layout_height="40dp"
+            android:layout_centerHorizontal="true"
+            android:scaleType="fitXY"
+            android:src="@drawable/ic_call_end"
+            android:background="@android:color/transparent" />
 
-    <TextView
-        android:id="@+id/end_text"
-        style="@style/label_bold"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_alignParentLeft="true"
-        android:layout_marginLeft="60dip"
-        android:layout_centerVertical="true"
-        android:textColor="@color/clock_white"
-        android:layout_below="@id/end_call_button"
-        android:text="@string/end_call"
-        android:textAlignment="center"
-        android:visibility="invisible" />
+        <TextView
+            android:id="@+id/ignore_text"
+            style="@style/label_bold"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_centerVertical="true"
+            android:textColor="@color/clock_white"
+            android:layout_below="@id/ignore_icon"
+            android:text="@string/ignore_call"
+            android:textAlignment="center" />
+    </RelativeLayout>
 
 </org.lineageos.flipflap.PhonePanel>
diff --git a/src/org/lineageos/flipflap/CircleView.java b/src/org/lineageos/flipflap/CircleView.java
index 315e08e..f5fc20b 100644
--- a/src/org/lineageos/flipflap/CircleView.java
+++ b/src/org/lineageos/flipflap/CircleView.java
@@ -27,6 +27,7 @@ import android.util.Log;
 import android.text.TextUtils;
 import android.view.View;
 import android.widget.ImageButton;
+import android.widget.RelativeLayout;
 import android.widget.TextView;
 
 public class CircleView extends FlipFlapView {
@@ -39,9 +40,9 @@ public class CircleView extends FlipFlapView {
     private PhonePanel mPhonePanel;
     private ImageButton mAlarmSnoozeButton;
     private ImageButton mAlarmDismissButton;
-    private ImageButton mAnswerCallButton;
-    private ImageButton mIgnoreCallButton;
-    private ImageButton mEndCallButton;
+    private RelativeLayout mAnswerCallButton;
+    private RelativeLayout mIgnoreCallButton;
+    private TextView mEndCallText;
     private TextView mIncomingCallName;
     private TextView mIncomingCallNumber;
     private boolean mRinging;
@@ -62,9 +63,9 @@ public class CircleView extends FlipFlapView {
 
         mIncomingCallName = (TextView) findViewById(R.id.incoming_call_name);
         mIncomingCallNumber = (TextView) findViewById(R.id.incoming_call_number);
-        mAnswerCallButton = (ImageButton) findViewById(R.id.answer_button);
-        mIgnoreCallButton = (ImageButton) findViewById(R.id.ignore_button);
-        mEndCallButton = (ImageButton) findViewById(R.id.end_call_button);
+        mAnswerCallButton = (RelativeLayout) findViewById(R.id.answer_button);
+        mIgnoreCallButton = (RelativeLayout) findViewById(R.id.ignore_button);
+        mEndCallText = (TextView) findViewById(R.id.ignore_text);
         mAnswerCallButton.setOnClickListener(new OnClickListener() {
             @Override
             public void onClick(View view) {
@@ -77,12 +78,6 @@ public class CircleView extends FlipFlapView {
                 endCall();
             }
         });
-        mEndCallButton.setOnClickListener(new OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                endCall();
-            }
-        });
 
         mAlarmSnoozeButton = (ImageButton) findViewById(R.id.snooze_button);
         mAlarmDismissButton = (ImageButton) findViewById(R.id.dismiss_button);
@@ -138,12 +133,10 @@ public class CircleView extends FlipFlapView {
             mNextAlarmPanel.setVisibility(View.GONE);
             if (mRinging) {
                 mAnswerCallButton.setVisibility(View.VISIBLE);
-                mIgnoreCallButton.setVisibility(View.VISIBLE);
-                mEndCallButton.setVisibility(View.GONE);
+                mEndCallText.setText(getResources().getString(R.string.ignore_call));
             } else {
                 mAnswerCallButton.setVisibility(View.GONE);
-                mIgnoreCallButton.setVisibility(View.GONE);
-                mEndCallButton.setVisibility(View.VISIBLE);
+                mEndCallText.setText(getResources().getString(R.string.end_call));
             }
         } else if (mAlarmActive) {
             mDatePanel.setVisibility(View.VISIBLE);
diff --git a/src/org/lineageos/flipflap/RectangularView.java b/src/org/lineageos/flipflap/RectangularView.java
index cd5d407..1500c35 100644
--- a/src/org/lineageos/flipflap/RectangularView.java
+++ b/src/org/lineageos/flipflap/RectangularView.java
@@ -28,6 +28,7 @@ import android.util.Log;
 import android.text.TextUtils;
 import android.view.View;
 import android.widget.ImageButton;
+import android.widget.RelativeLayout;
 import android.widget.TextView;
 
 public class RectangularView extends FlipFlapView {
@@ -40,9 +41,9 @@ public class RectangularView extends FlipFlapView {
     private PhonePanel mPhonePanel;
     private ImageButton mAlarmSnoozeButton;
     private ImageButton mAlarmDismissButton;
-    private ImageButton mAnswerCallButton;
-    private ImageButton mIgnoreCallButton;
-    private ImageButton mEndCallButton;
+    private RelativeLayout mAnswerCallButton;
+    private RelativeLayout mIgnoreCallButton;
+    private TextView mEndCallText;
     private TextView mIncomingCallName;
     private TextView mIncomingCallNumber;
     private boolean mRinging;
@@ -69,9 +70,9 @@ public class RectangularView extends FlipFlapView {
 
         mIncomingCallName = (TextView) findViewById(R.id.incoming_call_name);
         mIncomingCallNumber = (TextView) findViewById(R.id.incoming_call_number);
-        mAnswerCallButton = (ImageButton) findViewById(R.id.answer_button);
-        mIgnoreCallButton = (ImageButton) findViewById(R.id.ignore_button);
-        mEndCallButton = (ImageButton) findViewById(R.id.end_call_button);
+        mAnswerCallButton = (RelativeLayout) findViewById(R.id.answer_button);
+        mIgnoreCallButton = (RelativeLayout) findViewById(R.id.ignore_button);
+        mEndCallText = (TextView) findViewById(R.id.ignore_text);
         mAnswerCallButton.setOnClickListener(new OnClickListener() {
             @Override
             public void onClick(View view) {
@@ -84,12 +85,6 @@ public class RectangularView extends FlipFlapView {
                 endCall();
             }
         });
-        mEndCallButton.setOnClickListener(new OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                endCall();
-            }
-        });
 
         mAlarmSnoozeButton = (ImageButton) findViewById(R.id.snooze_button);
         mAlarmDismissButton = (ImageButton) findViewById(R.id.dismiss_button);
@@ -152,12 +147,10 @@ public class RectangularView extends FlipFlapView {
             mPhonePanel.setVisibility(View.VISIBLE);
             if (mRinging) {
                 mAnswerCallButton.setVisibility(View.VISIBLE);
-                mIgnoreCallButton.setVisibility(View.VISIBLE);
-                mEndCallButton.setVisibility(View.GONE);
+                mEndCallText.setText(getResources().getString(R.string.ignore_call));
             } else {
                 mAnswerCallButton.setVisibility(View.GONE);
-                mIgnoreCallButton.setVisibility(View.GONE);
-                mEndCallButton.setVisibility(View.VISIBLE);
+                mEndCallText.setText(getResources().getString(R.string.end_call));
             }
         } else if (mAlarmActive) {
             if (mNeedsSmallView) {
-- 
2.17.1

